README.md

```diff
@@ -2665,24 +2664,39 @@ Other Style Guides
   <a name="semicolons--required"></a><a name="20.1"></a>
   - [21.1](#semicolons--required) **Yup.** eslint: [`semi`](http://eslint.org/docs/rules/semi.html) jscs: [`requireSemicolons`](http://jscs.info/rule/requireSemicolons)
 
+    > Why? When JavaScript encounters a line break without a semicolon, it uses a set of rules called [Automatic Semicolon Insertion](https://tc39.github.io/ecma262/#sec-automatic-semicolon-insertion) to determine whether or not it should regard that line break as the end of a statement, and (as the name implies) place a semicolon into your code before the line break if it thinks so. ASI contains a few eccentric behaviors, though, and your code will break if JavaScript misinterprets your line break. These rules will become more complicated as new features become a part of JavaScript. Explicitly terminating your statements and configuring your linter to catch missing semicolons will help prevent you from encountering issues.
+
     ```javascript
-    // bad
-    (function () {
-      const name = 'Skywalker'
-      return name
-    })()
+    // bad - raises exception
+    const luke = {}
+    const leia = {}
+    [luke, leia].forEach(jedi => jedi.father = 'vader')
+
+    // bad - raises exception
+    const foo = 2
+    ("No! That's impossible!".repeat(foo))
+
+    // bad - returns `undefined` instead of the value on the next line - always happens when `return` is on a line by itself because of ASI!
+    function foo() {
+      return
+        "search your feelings, you know it to be foo"
+    }
 
     // good
-    (function () {
-      const name = 'Skywalker';
-      return name;
-    }());
+    const luke = {};
+    const leia = {};
+    [luke, leia].forEach(jedi => {
```

Comment:
This example is fine for illustrative purposes but it's not typical nor great when a line is started by `[`. This array needs a name.

Reply:
Oh? Hmm... [Is that a new rule?](https://github.com/airbnb/javascript#arrow-functions)

Reply:
Oh yeah we should change both :P but that's for another PR I guess.